Adapter-static documentation: Github pages (#2135)
sveltejs/kit: packages/adapter-static/README.md

When operating in SPA mode, only pages that have the [`prerender`](https://kit.svelte.dev/docs#ssr-and-javascript-prerender) option set will be prerendered.

## Github Pages

> When building for github pages, make sure to update the [basepaths](https://kit.svelte.dev/docs#configuration) (see paths.base and paths.assets).

Reviewer: Perhaps an example valid config for GitHub pages?

> Also note that you will have to prevent github's provided jekyll to manage your site by putting an empty `.nojekyll` file in your static folder. If you do not want to disable jekyll, change the kit's `appdir` configuration option to `"app_"` or anything not starting by an underscore. For more information, see github's documentation about [configuring jekyll](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll#configuring-jekyll-in-your-github-pages-site).

Reviewer 2: ```suggestion
Also note that you will have to prevent Github's provided Jekyll from managing your site by putting an empty `.nojekyll` file in your static folder. If you do not want to disable Jekyll, change the kit's `appdir` configuration option to `"app_"` or anything not starting by an underscore. For more information, see Github's documentation about [configuring jekyll](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll#configuring-jekyll-in-your-github-pages-site).
```

When operating in SPA mode, only pages that have the [`prerender`](https://kit.svelte.dev/docs#ssr-and-javascript-prerender) option set will be prerendered.

> ## Github Pages

Reviewer 3: hate to be a cop but we should fix all these

```suggestion
## GitHub Pages
```

When building for Github Pages, make sure to update the [basepaths](https://kit.svelte.dev/docs#configuration) (see `paths.base` and `paths.assets`).

> Also note that you will have to prevent Github's provided Jekyll from managing your site by putting an empty `.nojekyll` file in your static folder. If you do not want to disable Jekyll, change the kit's `appdir` configuration option to `"app_"` or anything not starting by an underscore. For more information, see Github's documentation about [configuring jekyll](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll#configuring-jekyll-in-your-github-pages-site).

Reviewer 3: ```suggestion
You will have to prevent GitHub's provided Jekyll from managing your site by putting an empty `.nojekyll` file in your static folder. If you do not want to disable Jekyll, change the kit's `appDir` configuration option to `"app_"` or anything not starting with an underscore. For more information, see GitHub's [Jekyll documentation](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll#configuring-jekyll-in-your-github-pages-site).
```

> A config for Github Pages might look like:

Reviewer 3: ```suggestion
A config for Github Pages might look like the following:
```

```js
/** @type {import('@sveltejs/kit').Config} */
const config = {
	...
	kit: {
		...

> 		// Since your Github Pages' default url will be https://your-username.github.io/your-repo-name,
> 		// you'll have the change the basepaths.

Reviewer 3: ```suggestion
		// Since your GitHub Pages site will be served at https://your-username.github.io/your-repo-name,
		// you will need to change the base paths
```

> 		// Note that you may have to comment these in dev mode.

Reviewer 2: This seems like quite the caveat. I don't know if we'd want to promote this solution without making it a bit nicer to use first

Reviewer 2: paths are currently broken in dev mode. i'll try to review https://github.com/sveltejs/kit/pull/2145 and then we can see if that changes the situation

Author: other option is to detect the dev environment: 
```js
const isProduction = process.env.NODE_ENV === 'production';

...
    paths: {
        base: isProduction ? "project_name" : "",
    }
...
```

		paths: {

> 			base: '/your-repo-name',
> 			assets: '/your-repo-name',

Reviewer 3: `assets` is resolved relative to `base`, so it can be left as the default

```suggestion
			base: '/your-repo-name'
```

		},
		// If you are not using a .nojekyll file, change your appDir to something not starting with an underscore.

> 		// For example, instead of "_app", use "app_", "internal", etc.

Reviewer 2: ```suggestion
		// For example, instead of '_app', use 'app_', 'internal', etc.
```

> 		appDir: "internal",

Reviewer 3: ```suggestion
		appDir: 'internal'
```
